pkg/reconciler/eventtransform/eventtransform.go

```diff
@@ -146,8 +163,54 @@ func (r *Reconciler) reconcileJsonataTransformationService(ctx context.Context,
 	return nil
 }
 
-func (r *Reconciler) reconcileJsonataTransformationDeployment(ctx context.Context, expression *corev1.ConfigMap, transform *eventing.EventTransform) error {
-	expected := jsonataDeployment(ctx, r.configWatcher, expression, transform)
+func (r *Reconciler) reconcileJsonataTransformationCertificate(ctx context.Context, transform *eventing.EventTransform) (*cmapis.Certificate, error) {
+	if f := feature.FromContext(ctx); !f.IsStrictTransportEncryption() && !f.IsPermissiveTransportEncryption() {
+		return nil, r.deleteJsonataTransformationCertificate(ctx, transform)
+	}
+	expected := jsonataCertificate(ctx, transform)
+
+	curr, err := r.cmCertificateLister.Certificates(expected.GetNamespace()).Get(expected.GetName())
+	if apierrors.IsNotFound(err) {
+		created, err := r.createCertificate(ctx, transform, expected)
+		if err != nil {
+			return nil, err
+		}
+		if !transform.Status.PropagateJsonataCertificateStatus(created.Status) {
+			// Wait for Certificate to become ready before continuing.
+			return nil, controller.NewSkipKey("")
+		}
+		return created, nil
+	}
+	if err != nil {
+		return nil, fmt.Errorf("failed to get certificate %s/%s: %w", expected.GetNamespace(), expected.GetName(), err)
+	}
+	if equality.Semantic.DeepDerivative(expected.Spec, curr.Spec) &&
+		equality.Semantic.DeepDerivative(expected.Labels, curr.Labels) &&
+		equality.Semantic.DeepDerivative(expected.Annotations, curr.Annotations) {
+		if !transform.Status.PropagateJsonataCertificateStatus(curr.Status) {
+			// Wait for Certificate to become ready before continuing.
+			return nil, controller.NewSkipKey("")
+		}
+		return curr, nil
+	}
+	expected.ResourceVersion = curr.ResourceVersion
+	updated, err := r.updateCertificate(ctx, transform, expected)
+	if err != nil {
+		return nil, err
+	}
+	if !transform.Status.PropagateJsonataCertificateStatus(updated.Status) {
+		// Wait for Certificate to become ready before continuing.
+		return nil, controller.NewSkipKey("")
+	}
+	return updated, nil
+}
+
+func (r *Reconciler) reconcileJsonataTransformationDeployment(ctx context.Context, expression *corev1.ConfigMap, certificate *cmapis.Certificate, transform *eventing.EventTransform) error {
+	withCombinedTrustBundle := false
+	if isPresent, _ := eventingtls.CombinedBundlePresent(r.trustBundleConfigMapLister); isPresent {
```

Comment:
why this check here, for combined / trustBundle?

Reply:
This is to check if we need to add the combined bundle volume or not (which is then injected by SinkBinding)

Reply:
OK, for integration we have this, since we leverage existing API via `Containersource`.

thx

Reply:
for IntegrationSource this might be relevant, how are we reading trust bundles on the kamelet side?